src/Statement.py

```diff
@@ -72,3 +80,40 @@ def get_currency(self):
             return self._statement[self._config.get_booking_currency()]
         else:
             return "EUR"
+
+    @staticmethod
+    def __match_category(mapping, booking_type, value):
+        """
+        takes a dict of format {"regex": "compiled regex", "category": "category"} and returns the correct mapping for
+        the category.
+
+        :param mapping: dict of type {regex": "compiled regex", "category": "category"}
+        :param booking_type: string containing the relevant loan information to determine category of entry.
+        :param value: value of the transaction, only required to handle special cases for mintos premium discount
+
+        :return: category
+        """
+        category = ""
+        if mapping["regex"] and mapping["regex"].match(booking_type):
+            category = mapping["category"]
+        if category == "Undecided":
+            category = Statement.__handle_special_case_mintos_discount_premium(value)
+        return category
+
+    @staticmethod
+    def __handle_special_case_mintos_discount_premium(value):
+        """
+        This is currently a special case for the Mintos "discount/premium" secondary market transactions parsing,
+        where an entry might be a fee or an income depending on its sign.
+
+        :param value: how much money was returned/paid
+
+        :return: Zinsen if value >= 0; Gebühren if < 0; Ignored in any other case
+        """
+
+        if value >= 0:
+            return "Zinsen"
+        elif value < 0:
+            return "Gebühren"
+        else:
+            return "Ignored"
```

Comment:
Could you please add some test data in src/test/testdata/mintos.csv and update the unit test so that we verify the correct behaviour? :) That would be fantastic!

Reply:
Done :) But in my portfolio I only have cases for "Gebühren" so I am not 100% sure on the other if-branch.

Reply:
one fix outstanding for the black findings :) will have a look after I finish gaming.

Reply:
You mentioned that you have one file without any date added? Could you add that as well to the test data please? Just to validate also that side of the branch :)
And hm.. looks like the if and elif branches are covered... at least according to my local checks. Missing would be the last one. Any idea? Maybe set the value field to a string for testing?

Reply:
Done :) In fact the `else` branch is never executed, as casting something not a number to a float before (in `get_value`) fails anyway, so we don't need that case here